openedx/core/djangoapps/catalog/tests/factories.py

```diff
@@ -36,8 +36,10 @@ def generate_zulu_datetime():
     The catalog returns UTC datetimes formatted using Z, the zone designator
     for the zero UTC offset, not the +00:00 offset. For more, see
     https://en.wikipedia.org/wiki/ISO_8601#UTC.
+    Faker.date_time() now returns time with sub-seconds in %S.%f format 
+    but we don't need the sub_seconds precision, but only the whole seconds
     """
-    return fake.date_time().isoformat() + 'Z'
+    return fake.date_time().isoformat().split('.')[0] + 'Z'
```

Comment:
With `Faker>20.0.0` now the default `Faker.date_time()` returns timestamp with [sub_seconds precision](https://github.com/joke2k/faker/blob/master/CHANGELOG.md#v2000---2023-11-10) in the format `%S.%f` e.g. `1984-08-01T23:40:55.814740` which causes [test failures](https://github.com/openedx/edx-platform/actions/runs/6906445463/job/18791495662?pr=33737) but since we don't need the sub_Seconds so we parse the value before returning to the views for testing.